crates/alerter/src/main.rs

```diff
@@ -236,43 +288,76 @@ impl Alerter {
             // In this case we would alert immediately even though it could be the case that the
             // order wasn't matchable and just now became matchable again. We would wrongly assume
             // it has been matchable since t0 but we did not check this between now and then.
-            for (_, instant) in self.open_orders.iter_mut() {
+            for (_, instant) in &mut self.open_orders {
                 *instant = None;
             }
             return Ok(());
         }
-        for i in 0..self.open_orders.len() {
+
+        if let Some((order, _)) = matchable_orders
+            .into_iter()
+            .find(|(_, solvable)| *solvable > self.config.min_order_solvable_time)
+        {
+            let should_alert = match self.last_alert {
+                None => true,
+                Some(instant) => instant.elapsed() >= self.config.min_alert_interval,
+            };
+            if should_alert {
+                self.last_alert = Some(Instant::now());
+                self.alert(&order);
+            }
+
+            self.metrics.no_trades_but_matchable_order.set(1);
+        } else {
+            self.metrics.no_trades_but_matchable_order.set(0);
+        }
+
+        Ok(())
+    }
+
+    async fn matchable_orders(&mut self) -> Result<Vec<(Order, Duration)>> {
+        let mut result = Vec::new();
+        for (order, solvable_since) in &mut self.open_orders {
```

Comment:
With the expectation that the auction will be growing soon this might need to be a bit more sophisticated than querying zeroex for every order one after the other. But obviously out of scope for this PR.

Reply:
Good point. I also think we should MUX out to multiple DEX aggregators just in case.

Reply:
I don't know if this will work in practice. If we always check all orders then this function can take a really long time, and it errors if any individual request errors.